### PR TITLE
Skip tests timing out on Mac

### PR DESCRIPTION
**Pull request by zivkan**
## Bug

Tracking: https://github.com/NuGet/Home/issues/11321

Regression? No

## Description

Something appears to have changed on the Mac CI agents, and now all the commands to add certificates to the cert store are failing/taking a very long time, causing our pipeline to time out.

I searched for all instances of `SigningTestUtility.GenerateTrustedTestCertificate` and skipped those tests on Mac.

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [x] Test exception: skipping tests causing problems
  - **OR**
  - [ ] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A

### CODE REVIEW

File: test/TestUtilities/Test.Utility/PlatformXunitAttributes/PlatformTheoryAttribute.cs [hunk 1]
```diff
@@ -29,6 +29,14 @@ public override string Skip
...
                 }
 
+                if (string.IsNullOrEmpty(skip))
+                {
+                    if (CIOnly && !XunitAttributeUtility.IsCI)
+                    {
+                        skip = "This test only runs on the CI. To run it locally set the env var CI=true";
+                    }
+                }
+
```
Comment (heng-liu) [inline test/TestUtilities/Test.Utility/PlatformXunitAttributes/PlatformTheoryAttribute.cs:39]: Shall we rebase from dev as it's already merged into dev from another PR? Thanks.

**Review by heng-liu — COMMENTED** [review]
Thanks for the PR!
Might need to add reference to `NuGet.Common` to be able to use `RuntimeEnvironmentHelper`.
Otherwise, it looks good to me.

**Review by erdembayar — APPROVED** [review]

File: test/TestUtilities/Test.Utility/PlatformXunitAttributes/PlatformTheoryAttribute.cs (hunk 1 above)
Comment (zivkan) [reply test/TestUtilities/Test.Utility/PlatformXunitAttributes/PlatformTheoryAttribute.cs:39]: I did rebase. That other PR added CIOnly to PlatformFactAttribute. THis is PlatformTheoryAttribute.

Comment (heng-liu) [reply test/TestUtilities/Test.Utility/PlatformXunitAttributes/PlatformTheoryAttribute.cs:39]: Sorry, I didn't notice the difference.

**Review by heng-liu — APPROVED** [review]

File: test/TestUtilities/Test.Utility/PlatformXunitAttributes/PlatformTheoryAttribute.cs (hunk 1 above)
Comment (zivkan) [reply test/TestUtilities/Test.Utility/PlatformXunitAttributes/PlatformTheoryAttribute.cs:39]: Neither did I until Intellisense gave me red squiggles and I had to figure out why 😁

**Comment by heng-liu** [comment]
Looks like there are more tests need to be skipped..
e.g. tests in DotnetSignTests 
Any tests need to run the following method will fail..
https://github.com/NuGet/NuGet.Client/blob/dev/test/TestUtilities/Test.Utility/Signing/TrustedTestCert.cs#L124

**Pull request merged**